src/FSharp.FGL.IO/GDF.fs

```diff
@@ -0,0 +1,157 @@
+        let vertexId = 
+            infos.TryFind "name"
+
+        match vertexId with
+        | Some id -> (reconstructTypeInfo id),(infos)
+        | None -> failwithf "vertex %O does not contain any identifier" infos
+
+    //Takes header and a line of string, applies getTypeInfoMapper on header, and applies these functions on the line contents. Builds a Map(header,TypeInfo), which is used for getVertedOfInfos 
+    let private getVertexByHeader (header:string) (line:string) = 
+        header.Split ','
+        |> Array.map (getTypeInfoMapper)
+        |> fun f -> 
+            line 
+            |> splitElementInfos
+            |> Array.map2 (fun f x -> f x) f
+            |> Array.map2 (fun a b -> a,b) (headerTrim header)
+            |> Map.ofArray
+            |> getVertexOfInfos
+
+    //Searches for the "node1" and "node2" identifier of the Edge Map(header,TypeDefiniton), if found returns (string node1)(string node2),Map tripel       
+    let private getEdgeOfinfos (infos : Map<string,TypeInfo>) : LEdge<string,Map<string,TypeInfo>> =
+        let vertexId1 = 
+            infos.TryFind "node1" 
+
+        let vertexId2 =
+            infos.TryFind "node2" 
+
+        match vertexId1,vertexId2 with
+        | Some id1,Some id2 -> reconstructTypeInfo id1,reconstructTypeInfo id2,infos
+        | Some _ , None     -> failwithf "edge %O does not contain a target vertex" infos
+        | None, Some _      -> failwithf "edge %O does not contain a source vertex" infos
+        | None,None         -> failwithf "edge %O does not contain any vertex ids" infos
+
+    //Takes header and a line of string, applies getTypeInfoMapper on header, and applies these functions on the line contents. Builds a Map(header,TypeInfo), which is used for getEdgeOfInfos 
+    let private getEdgeByHeader (header:string) (line:string) = 
+        header.Split ','
+        |> Array.map (getTypeInfoMapper)
+        |> fun f -> 
+            line 
+            |>splitElementInfos
+            |> Array.map2 (fun f x -> f x) f
+            |> Array.map2 (fun a b -> a,b) (headerTrim header)
+            |> Map.ofArray
+            |> getEdgeOfinfos
+
+    //activePatternMatching, recognizes VertexHeader and returns getVertexByHeader applied on the header
+    let private (|VertexHeader|_|) (s:string) =
+        let r = System.Text.RegularExpressions.Regex.Match(s,"(?<=nodedef>).*")
+        if r.Success then Some(getVertexByHeader r.Value)
+        else None
+
+    //activePatternMatching, recognizes EdgeHeader and returns getEdgeByHeader applied on the header
+    let private (|EdgeHeader|_|) (s:string) =
+        let r = System.Text.RegularExpressions.Regex.Match(s,"(?<=edgedef>).*")
+        if r.Success then Some(getEdgeByHeader r.Value)
+        else None
+    
+
+    //Takes string [], recognizes if data belongs to 'Vertex or 'Edge, and returns a tupel of vertices and edges
+    let private getVertexAndEdgeFromData (data: string []) =
```

Comment:
You could rename this function to `fromArray` and ditch the `fromSeq` function below, as it does nothing really, beside calling this function